modules/lava.lua

```diff
@@ -0,0 +1,539 @@
+		     { 150, 0.25, 5    },
+		     { -20, 1.00, 5*60 },
+		     { 180, 0.50, 60   },
+		     { 240, 0.20, 10   } }
+end
+
+]]
+
+if mapLavaConfig and not voidWaterMap then
+	isLavaMap = true
+
+	diffuseEmitTex = mapLavaConfig.diffuseEmitTex or diffuseEmitTex
+	normalHeightTex = mapLavaConfig.normalHeightTex or normalHeightTex
+
+	level = mapLavaConfig.level or level
+	grow = mapLavaConfig.grow or grow
+	damage = mapLavaConfig.damage or damage
+	damageMode = mapLavaConfig.damageMode or damageMode
+	damageMinHealth = mapLavaConfig.damageMinHealth or damageMinHealth
+	if mapLavaConfig.damageFeatures ~= nil then
+		damageFeatures = mapLavaConfig.damageFeatures
+	end
+	uvScale = mapLavaConfig.uvScale or uvScale
+	colorCorrection = mapLavaConfig.colorCorrection or colorCorrection
+	losDarkness = mapLavaConfig.losDarkness or losDarkness
+	swirlFreq = mapLavaConfig.swirlFreq or swirlFreq
+	swirlAmp = mapLavaConfig.swirlAmp or swirlAmp
+	specularExp = mapLavaConfig.specularExp or specularExp
+	shadowStrength = mapLavaConfig.shadowStrength or shadowStrength
+	coastWidth = mapLavaConfig.coastWidth or coastWidth
+	coastColor = mapLavaConfig.coastColor or coastColor
+	coastLightBoost = mapLavaConfig.coastLightBoost or coastLightBoost
+
+	parallaxDepth = mapLavaConfig.parallaxDepth or parallaxDepth
+	parallaxOffset = mapLavaConfig.parallaxOffset or parallaxOffset
+
+	fogColor = mapLavaConfig.fogColor or fogColor
+	fogFactor = mapLavaConfig.forFactor or fogFactor
+	fogHeight = mapLavaConfig.fogHeight or fogHeight
+	fogAbove = mapLavaConfig.fogAbove or fogAbove
+	if mapLavaConfig.fogEnabled ~= nil then
+		fogEnabled = mapLavaConfig.fogEnabled
+	end
+	fogDistortion = mapLavaConfig.fogDistortion or fogDistortion
+
+	tideAmplitude = mapLavaConfig.tideAmplitude or tideAmplitude
+	tidePeriod = mapLavaConfig.tidePeriod or tidePeriod
+	tideRhym = mapLavaConfig.tideRhym or tideRhym
+	effectDamage = mapLavaConfig.effectDamage or effectDamage
+	if mapLavaConfig.effectBurst ~= nil then
+		effectBurst = mapLavaConfig.effectBurst
+	end
+	effectBurstSounds = mapLavaConfig.effectBurstSounds or effectBurstSounds
+	if mapLavaConfig.ambientSounds ~= nil then
+		ambientSounds = mapLavaConfig.ambientSounds
+	end
+
+elseif string.find(mapName, "stronghold") then
+	isLavaMap = true
+	level = 20
+	grow = 0
+
+	damage = 25 -- damage per second
+	tideAmplitude = 3
+	tidePeriod = 95
+	diffuseEmitTex = "LuaUI/images/lava/lava7_diffuseemit.dds"
+	normalHeightTex = "LuaUI/images/lava/lava7_normalheight.dds"
+	losDarkness = 0.7
+	colorCorrection = "vec3(1.1, 1.0, 0.88)"
+	shadowStrength = 1.0 -- how much light a shadowed fragment can recieve
+	coastColor = "vec3(2.2, 0.4, 0.0)"
+	coastLightBoost = 0.7
+	coastWidth = 36.0
+	fogFactor = 0.08 -- how dense the fog is
+	fogColor = "vec3(2.0, 0.31, 0.0)"
+	fogHeight = 85
+	fogAbove = 0.18
+	uvScale = 0.5
+	swirlFreq = 0.017
+	swirlAmp = 0.0024
+
+	tideRhym = { { 19, 0.3, 5*6000 } }
+
+elseif string.find(mapName, "incandescence") then
```

Comment:
Will do that, also maybe make some changes for compatibility with z-k, I like the idea of loading the config from mapconfig/lava_config.lua instead of mapinfo.lua for example, that way the file we provide under whatever/Configs/LavaConf/MapXX.lua can be the same as can be included into the mapfile.

I see they did add some of the things I added, albeit with different names, also they have slightly different names for some of the vars (like lOSdarkness instead of losDarkness), not sure how worth it it is to improve map compatibility though, since there might be other areas where the two projects diverged already.

Reply:
Ok, done. I did it a bit different than z-k to allow for config to be loaded without the version name, otherwise it would break with any small change by the map creator. I still allow loading with version too though, but provide all configs without version in the name. Also I prioritize map provided config unless the game sets overrideMap in its map specific lava config.

Besides that, regarding compatibility with z-k I think it can be a bit hopeless so I load the config from a different path than them: mapconfig/lava.lua, this way maps could have both configs to support both styles.

Anyways let me know if you think more direct compatibility is wanted, but tbh I really don't like some of the capitalization they chose, like lOSdarkness, uVscale, tideamplitude, tideperiod. It's just an esthetic thing, so if compatibility is preferred I could use the capitalization they chose and also change a few names. Still, the implementation is likely to diverge so even if we provide "compatible" configs it's likely the lava would be broken in subtle ways.

That's why I chose to read a different file.